packages/frontend/web-ui/src/auth/hooks/useRegisterConfirm.spec.js

```diff
@@ -0,0 +1,281 @@
+        configurable: true,
+      });
+    });
+
+    describe('when called, and httpClient.updateUserMe() returns an OK response', () => {
+      let result;
+      let status;
+      let errorMessage;
+
+      beforeAll(async () => {
+        tokenFixture = 'jwt token fixture';
+
+        useDispatch.mockReturnValue(dispatchMock);
+        useSelector.mockImplementation(() => ({
+          token: tokenFixture,
+          errorMessage: authErrorMessageFixture,
+        }));
+
+        httpClient.updateUserMe.mockReturnValueOnce({
+          headers: {},
+          body: {
+            active: true,
+            id: 'id',
+            name: 'name',
+          },
+          statusCode: 200,
+        });
+
+        buildSerializableResponse.mockImplementation((response) => ({
+          body: response.body,
+          statusCode: response.statusCode,
+        }));
+
+        await act(() => {
+          result = renderHook(() => useRegisterConfirm()).result;
+        });
+
+        status = result.current.status;
+        errorMessage = result.current.errorMessage;
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+        jest.clearAllMocks();
+      });
+
+      it('should called useDispatch()', () => {
+        expect(useDispatch).toHaveBeenCalled();
+      });
+
+      it('should called useSelector() ', () => {
+        expect(useSelector).toHaveBeenCalled();
+      });
+
+      it('should return a fulfilled status', () => {
+        expect(status).toBe(STATUS_FULFILLED);
+      });
+
+      it('should return a null error message', () => {
+        expect(errorMessage).toBeNull();
+      });
+    });
+
+    describe('when called, and httpClient.updateUserMe() returns a non OK response', () => {
+      let result;
+      let status;
+      let errorMessage;
+
+      beforeAll(async () => {
+        tokenFixture = 'jwt token fixture';
+        authErrorMessageFixture = UNEXPECTED_ERROR_MESSAGE;
+
+        useDispatch.mockReturnValue(dispatchMock);
+        useSelector.mockImplementation(() => ({
+          token: tokenFixture,
+          errorMessage: authErrorMessageFixture,
+        }));
+
+        httpClient.updateUserMe.mockReturnValueOnce({
+          headers: {},
+          body: {
+            code: null,
+            description: 'Error Fixture',
+            parameters: {},
+          },
+          statusCode: 401,
+        });
+
+        buildSerializableResponse.mockImplementation((response) => ({
+          body: response.body,
+          statusCode: response.statusCode,
+        }));
+
+        await act(() => {
+          result = renderHook(() => useRegisterConfirm()).result;
+        });
+        status = result.current.status;
+        errorMessage = result.current.errorMessage;
+      });
+
+      it('should called useDispatch()', () => {
+        expect(useDispatch).toHaveBeenCalled();
+      });
+
+      it('should called useSelector() ', () => {
+        expect(useSelector).toHaveBeenCalled();
```

Comment:
`toHaveBeenCalledTimes` is more accurate.

In addition, assert mock arguments with `toHaveBeenCalledWith`.

Reply:
With `toHaveBeenCalledTimes` was called three times because the component does 3 iterations until it is rendered. I have done the `toHaveBeenCalledWith`.